### ISSUE TITLE
Analyzer fuzz crash: invoking const constructor 

### ISSUE THREAD

**Issue by MichaelRFairhurst** [dart-lang/sdk#38204]
```dart
class C<T> {
  const C(this.x);
  f() {
    const C(null);
  }
}
```

Note: this only fails in summary1, not summary2, making it potentially lower priority. Though this does seem to be really easy to hit (and has been hit in #37996), so might be worth the effort of fixing in the mean time.

```
NoSuchMethodError: The getter 'type' was called on null.
Receiver: null
Tried calling: type
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      VariableMember.type (package:analyzer/src/dart/element/member.dart:1053:61)
#2      ConstantEvaluationEngine.evaluateConstructorCall (package:analyzer/src/dart/constant/evaluation.dart:626:40)
#3      ConstantVerifier.visitInstanceCreationExpression (package:analyzer/src/dart/constant/constant_verifier.dart:134:27)
#4      InstanceCreationExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:6180:15)
#5      ExpressionStatementImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:3770:18)
#6      RecursiveAstVisitor.visitExpressionStatement (package:analyzer/dart/ast/visitor.dart:833:10)
#7      ExpressionStatementImpl.accept (package:analyzer/src/dart/ast/ast.dart:3766:49)
#8      NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7670:20)
#9      BlockImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:1115:17)
#10     RecursiveAstVisitor.visitBlock (package:analyzer/dart/ast/visitor.dart:665:10)
```

**Comment by bwilkerson** [comment]
I don't think there's any "mean time" any more. Feel free to re-open if I'm wrong.

**Comment by MichaelRFairhurst** [comment]
Hmm. This does indeed seem to fail in summary2. Just hit this there. Not sure why I had originally thought otherwise.

**Comment by scheglov** [comment]
Also fixed by c6b0ae1fdab360f08b7966797506dae6629551fd.
